tests/core/mixins/treeitem.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2015 Evernote Corporation
+#
+# This file is part of Pootle.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, see <http://www.gnu.org/licenses/>.
+
+import pytest
+
+from pootle_app.models import Directory
+from pootle_project.models import Project
+from pootle_store.models import Store
+from pootle_translationproject.models import TranslationProject
+
+
+@pytest.mark.django_db
+def test_get_children(tutorial, afrikaans):
+    def _all_children_are_directories_or_stores(item):
+        for child in item.children:
+            if isinstance(child, Directory):
+                _all_children_are_directories_or_stores(child)
+            else:
+                assert isinstance(child, Store)
+
+    for tp in tutorial.children:
+        assert isinstance(tp, TranslationProject)
+        _all_children_are_directories_or_stores(tp)
+
+    for tp in afrikaans.children:
+        assert isinstance(tp, TranslationProject)
+        _all_children_are_directories_or_stores(tp)
+
+
+@pytest.mark.django_db
+def test_get_parents(af_tutorial_subdir_po, af_tutorial_po, afrikaans_tutorial, afrikaans, tutorial):
+    def _check_parents(item):
```

Comment:
Don't you think the function makes the test more difficult to understand? All of the type checking in the first branching is redundant, because you already know the type of object you are passing:
* `af_tutorial_subdir_po` and `af_tutorial_po` are `Store`, so you know you should expect the parents to be either `Directory` or `TranslationProject`.
* The rest are directories, and only the second branch applies.
* The third branch never happens.